neonvm/controllers/virtualmachine_controller.go

```diff
@@ -412,33 +415,38 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 					return err
 				}
 				// compare guest spec and count of plugged
-				if *virtualmachine.Spec.Guest.CPUs.Use > int32(len(cpusPlugged)) {
+				if int32(virtualmachine.Spec.Guest.CPUs.Use.Value()) > int32(len(cpusPlugged)) {
```

Comment:
+1. BTW, is there some special reason why it was done this way initially (with int32 and not int that `Len` already returns)?

Reply:
I'd imagine that `Spec.Guest.CPUs.Use` was `*int32` instead of `*int` so that the size of the integer wasn't platform-dependent (maybe kubebuilder complains?). But I don't know for sure

Reply:
Yeah, but isn't it better to upcast to not lose data? Like int(int32(1)) is better than int32(int(1))

Reply:
Ah, sorry yeah. I think _in general_ it's better, but realistically we're never going to have more than 2^32 CPUs plugged

So, I'm not @cicdteam, but I know that personally, I'd rather make the more unknown type clear (`len` is broadly known, but someone reading may not know that `CPUs.Use` is `int32`).

Either way, should be fine :smile: